packages/catalog-client/src/types.ts

```diff
@@ -40,6 +40,7 @@ export type AddLocationRequest = {
   type?: string;
   target: string;
   dryRun?: boolean;
+  presence?: 'optional' | 'required';
```

Comment:
This request type is super weird at this point in the repo (through no fault of yours). I wonder why the type and target were ever optional. And there's already a `LocationSpec` type in `catalog-model`, and a corresponding validation schema - where type and target are required of course.

This type should probably rather be

```ts
export type AddLocationRequest = {
  spec: LocationSpec;
  dryRun?: boolean;
}
```